src/operator/optimizer_op-inl.h

```diff
@@ -225,7 +225,7 @@ struct MultiSGDKernelParam {
 template <typename MPDType, bool has_momentum, bool has_mixed_precision>
 struct MultiSGDKernel {
   template<typename DType>
-  MSHADOW_XINLINE static void Map(int i, const MultiSGDKernelParam<DType, MPDType>& param,
+  MSHADOW_XINLINE static void Map(index_t i, const MultiSGDKernelParam<DType, MPDType>& param,
     const OpReqType req) {
     for (int index = 0; index < param.count; ++index) {
       if ((size_t)i < param.sizes[index]) {
```

Comment:
Is it safe to convert `index_t` to `size_t`?

Reply:
um ya. param.sizes returns size_t so has to be checked that way shouldn't it?

Reply:
On some 64bit system, `size_t` may be smaller than `int64_t` (`index_t` in this case). That's why I have concern.

Reply:
@ChaiBapchya size_t is platform dependent. It is safer to static_cast param.sizes[index] to index_t. When large tensor is not enabled and param.sizes[index] is greater than 2^32 we already have error check at python level.

Reply:
Makes sense. Addressed!

Reply:
> @ChaiBapchya size_t is platform dependent. It is safer to static_cast param.sizes[index] to index_t. When large tensor is not enabled and param.sizes[index] is greater than 2^32 we already have error check at python level.

@apeforest @szhengac sinze size_t is platform dependent even if we build with large tensor flag int64_t won't be supported there if size_t is not uint_64. My concern here is: "if the value of (param.sizes[index])is between 2^63 and 2^64-1 then type casting it to int64_t will corrupt the value, which is incorrect. Let me know if you need more details to understand this use case.

Reply:
@access2rohit I think this is a problem for non 64-bit platform, not only to this line of code. A better solution could be adding an assert in the front end when tensor size/index is greater than int32 and platform is not 64-bit. What do you think?